### ISSUE TITLE
Wrong text placement, when dpiY <> 96

### ISSUE THREAD

**Issue by butlnor** [vvvv/SVG#214]
I am using Svg library for generating PDF (via rendering SVG images to Metafile and embedding them with iTextSharp) and also for the direct printing.

However, PDF generating is perfect, while printing doesn't properly position the texts. I've been analyzing the Svg source files and the problem is in method Svg.GdiFontDefn.Ascent(ISvgRenderer renderer).
This method does some computation, but the returned float value is only correct, when dpiY of the Graphics is exactly 96. Since, luckily, Metafile's default DPI is also 96, my PDF generating is ok, However, as (my) default printing DPI is 300, all the texts are rendered erroneously.

For the test, I attached a "SvgError.svg" (zipped) with 3 colorful rectangles and texts. When printing (to the file) such an image, you get something like in the attached "output.png", where the texts are located too far up from their proper location.

[SvgError.zip](https://github.com/vvvv/SVG/files/87866/SvgError.zip)

![output](https://cloud.githubusercontent.com/assets/11555256/12274197/1a9a397a-b96a-11e5-82bc-19584fb7d65f.png)

Source code for easier testing:

``` csharp
using System;
using System.Windows.Forms;
using System.Drawing.Printing;
using Svg;
using Svg.Transforms;

namespace SvgError {
    public class App {
        [STAThread]
        public static void Main(string[] args) {
            PrintDocument printDoc = new PrintDocument();
            printDoc.PrintPage += printDoc_PrintPage;
            PrintDialog printDialog = new PrintDialog();
            printDialog.Document = printDoc;

            if (printDialog.ShowDialog(null) == DialogResult.OK) {
                printDoc.Print();
            }
        }

        private static void printDoc_PrintPage(object sender, PrintPageEventArgs e) {
            SvgDocument svgDoc = SvgDocument.Open<SvgDocument>("SvgError.svg");
            svgDoc.Draw(e.Graphics);
        }
    }
}

```

**Comment by butlnor** [comment]
Currently I am using a workaround by writing a SvgRendererWrapper class, which implements ISvgRenderer, but delegates all the methods and the properties to the underlying SvgRenderer, except the DpiY property, in which I am always returning a selected constant value.

However, this is just an ugly workaround, as the Svg library is supposed to work correctly.

```csharp
    public class SvgRendererWrapper : ISvgRenderer {
        private ISvgRenderer parent;
        private float dpiY;

        public SvgRendererWrapper(ISvgRenderer parent, float dpiY) {
            this.parent = parent;
            this.dpiY = dpiY;
        }

        public float DpiY {
            get {
                return this.dpiY;
            }
        }

        // all other methods and properties just delegate to "parent".
    }

**Comment by Cian-Chambliss** [comment]
I Ran into the same problem - this issue is that the Graphics context treats printer output as 100dpi, yet  graphics.DpiY returns the actual DPI of the printer device (generally something like 600dpi).

the Text logic is the only consumer of the DpiY - it uses this when calculating the baseline, which ends up being 6x more than it should if the printer is 600dpi.

**Comment by mrbean-bremen** [comment]
Shall be fixed now with #356 - closing.
